VogtObserver/Program.cs

```diff
@@ -1,12 +1,37 @@
 ﻿using System;
+using System.Threading;
 
 namespace VogtObserver
 {
     class Program
     {
         static void Main(string[] args)
         {
-            
+            Generator generator = new Generator();
+            Date dateTime = new Date(generator);
+            Sum sum = new Sum(generator);
+
+            Run(generator, dateTime, sum);
+        }
+
+        public static void Run(Generator generator, Date dateTime, Sum sum)
+        {
+            for (int i = 1; i <=20 ; i++)
+            {
+                try
+                {
+                    Console.Clear();
+                    generator.GetRandomInt();
+                    dateTime.Update();
+                    sum.Update();
```

Comment:
I think this is a similar issue to what I pointed out before about the other observer. You generate a random int and then tell the sum (the observer) that it needs to update. No one is actually observing anything. You get the random int and then tell the sum class "go update!".

Using the observer pattern should abstract the fact that the observer needs to be updated. You should be ADDING sum as the observer to the generator class.

```c#
public static void Run(Generator generator, Date dateTime, Sum sum)
        {
            // add observers
            generator.Add(dateTime);
            generator.Add(sum);

            for (int i = 1; i <=20 ; i++)
            {
                try
                {
                    Console.Clear();
                    generator.GetRandomInt();
                    // nothing left to do because the observers should have been notified by the generator using the pattern.
                    // therefore, you dont have to call .update()! by adding them as observers, update() will be called internally
                    Thread.Sleep(2000);
                }
                catch
                {
                    Console.WriteLine("Error!");
                }

            }
```